Analysis/Tasks/PWGHF/HFXicToPKPiCandidateSelector.cxx

```diff
@@ -0,0 +1,447 @@
+                                            {0.400, 0.4, 0.4, 0.4, 0.05, 0.09, 0.005, 0.},  /* 3<pt<4   */
+                                            {0.400, 0.4, 0.4, 0.4, 0.05, 0.09, 0.005, 0.},  /* 4<pt<5   */
+                                            {0.400, 0.4, 0.4, 0.4, 0.05, 0.09, 0.005, 0.},  /* 5<pt<6   */
+                                            {0.400, 0.4, 0.4, 0.4, 0.05, 0.09, 0.005, 0.},  /* 6<pt<8   */
+                                            {0.400, 0.4, 0.4, 0.4, 0.05, 0.09, 0.005, 0.},  /* 8<pt<12  */
+                                            {0.400, 0.4, 0.4, 0.4, 0.05, 0.09, 0.005, 0.},  /* 12<pt<24 */
+                                            {0.400, 0.4, 0.4, 0.4, 0.05, 0.09, 0.005, 0.}}; /* 24<pt<36 */
+
+/// Struct for applying Xic selection cuts
+struct HFXicToPKPiCandidateSelector {
+
+  Produces<aod::HFSelXicToPKPiCandidate> hfSelXicToPKPiCandidate;
+
+  Configurable<bool> d_FilterPID{"d_FilterPID", true, "Bool to use or not the PID at filtering level"};
+
+  Configurable<double> d_pTCandMin{"d_pTCandMin", 0., "Lower bound of candidate pT"};
+  Configurable<double> d_pTCandMax{"d_pTCandMax", 36., "Upper bound of candidate pT"};
+
+  Configurable<double> d_pidTPCMinpT{"d_pidTPCMinpT", 0.15, "Lower bound of track pT for TPC PID"};
+  Configurable<double> d_pidTPCMaxpT{"d_pidTPCMaxpT", 1., "Upper bound of track pT for TPC PID"};
+  Configurable<double> d_pidTOFMinpT{"d_pidTOFMinpT", 0.5, "Lower bound of track pT for TOF PID"};
+  Configurable<double> d_pidTOFMaxpT{"d_pidTOFMaxpT", 4., "Upper bound of track pT for TOF PID"};
+
+  Configurable<double> d_TPCNClsFindablePIDCut{"d_TPCNClsFindablePIDCut", 70., "Lower bound of TPC findable clusters for good PID"};
+  Configurable<double> d_nSigmaTPC{"d_nSigmaTPC", 3., "Nsigma cut on TPC only"};
+  Configurable<double> d_nSigmaTPCCombined{"d_nSigmaTPCCombined", 5., "Nsigma cut on TPC combined with TOF"};
+  Configurable<double> d_nSigmaTOF{"d_nSigmaTOF", 3., "Nsigma cut on TOF only"};
+  Configurable<double> d_nSigmaTOFCombined{"d_nSigmaTOFCombined", 5., "Nsigma cut on TOF combined with TPC"};
+
+  /// Gets corresponding pT bin from cut file array
+  /// \param candpT is the pT of the candidate
+  /// \return corresponding bin number of array
+  template <typename T>
+  int getpTBin(T candpT)
+  {
+    double pTBins[npTBins + 1] = {0, 1., 2., 3., 4., 5., 6., 8., 12., 24., 36.};
+    if (candpT < pTBins[0] || candpT >= pTBins[npTBins]) {
+      return -1;
+    }
+    for (int i = 0; i < npTBins; i++) {
+      if (candpT < pTBins[i + 1]) {
+        return i;
+      }
+    }
+    return -1;
+  }
+
+  /// Selection on goodness of daughter tracks
+  /// \note should be applied at candidate selection
+  /// \param track is daughter track
+  /// \return true if track is good
+  template <typename T>
+  bool daughterSelection(const T& track)
+  {
+    if (track.sign() == 0) {
+      return false;
+    }
```

Comment:
If I see it correctly, the track charge sign is checked only here.
Since the track charge can never be 0, this condition is useless and can be removed.

Reply:
At the moment yes, there are no other places where the charge is checked.
For the moment, I followed your instructions.

A good thing would be to check the sign of the 3 daughter together, verifying if they are like +-+ or -+- (in other words: avoiding to build triplets with +++, ---, --+ and so on).
But I prefer to think at it for the next PR eventually.

Reply:
Where do you get these triplets? From the skimming you get only +-+ or -+-.

Reply:
... ok, then what I was thinking is already done. Thanks for the clarification.